README.md

```diff
@@ -1,63 +1,307 @@
+
+<br>
+
+## A Look at Structures Used in Wrangler
 ```golang
 type FooController interface {
 	FooClient
 
+	// OnChange resgisters a handler that will run whenever an object of the matching resource type is updated. This function accepts a sync function specifically generated for the object type and then wraps the function in a function that is compatible with AddGenericHandler. It then uses AddGenericHandler to register the wrapped function.
 	OnChange(ctx context.Context, name string, sync FooHandler)
+	// OnRemove registers a handler that will run whenever an object of the matching resource type is removed. This function accepts a sync function specifically generated for the object type and then wraps the function in a function that is compatible with AddGenericRemoveHandler. It then uses AddGenericRemoveHandler to register the wrapped function.
 	OnRemove(ctx context.Context, name string, sync FooHandler)
+	// Enqueue will rerun all handlers registered to the object's type against the object.
 	Enqueue(namespace, name string)
 
+	// Cache returns a locally maintained cache that can be used for get and list requests.
 	Cache() FooCache
 
+	// Informer returns an informer for the resource type
 	Informer() cache.SharedIndexInformer
+	// GroupVersionKind returns the API group, version, and Kind of the resource type the controller is for
 	GroupVersionKind() schema.GroupVersionKind
 
+	// AddGenericHandler registers the handler function for the controller.
 	AddGenericHandler(ctx context.Context, name string, handler generic.Handler)
+	// AddGenericRemoveHandler registers a handler that will happen when an object of the controller's resource type is removed
 	AddGenericRemoveHandler(ctx context.Context, name string, handler generic.Handler)
 	Updater() generic.Updater
 }
 
 type FooClient interface {
+	// Create creates a new instance of resource type in kubernetes
 	Create(*v1alpha1.Foo) (*v1alpha1.Foo, error)
+	// Update updates the given object in kubernetes
 	Update(*v1alpha1.Foo) (*v1alpha1.Foo, error)
+	// Status of type's CRD must be a subresource for this method to be generated. Only updates
+	// status and does not trigger OnChange handlers. TODO: confirm this.
 	UpdateStatus(*v1alpha1.Foo) (*v1alpha1.Foo, error)
+	// Delete deletes the given object in kubernetes
 	Delete(namespace, name string, options *metav1.DeleteOptions) error
+	// Get gets the object of the given name and namespace in kubernetes
 	Get(namespace, name string, options metav1.GetOptions) (*v1alpha1.Foo, error)
+	// List lists all the objects matching the given namespace for the resource type
 	List(namespace string, opts metav1.ListOptions) (*v1alpha1.FooList, error)
+	// Watch returns a channel that will stream objects as they are created, removed, or updated
 	Watch(namespace string, opts metav1.ListOptions) (watch.Interface, error)
+	// Patch will create or updates the given object. Patch type specifies the how the data is structured, i.e. json
```

Comment:
> create

is this true?

Reply:
I think this is conditionally true. If the patch type is `application/apply-patch+yaml`, then yes. The second part about is incorrect though. I've updated to give a more general description of patch, fixed the inaccuracies, and provided some additional documentation links since this is a more complicated method which does not seem well understood by the Kubernetes community in general.